matrix-sdk-android/src/test/java/org/matrix/android/sdk/internal/session/room/aggregation/poll/PollAggregationProcessorTest.kt

```diff
@@ -0,0 +1,301 @@
+        type = EventType.POLL_RESPONSE.first(),
+        eventId = AN_EVENT_ID,
+        originServerTs = 1652435922563,
+        senderId = A_USER_ID_1,
+        roomId = A_ROOM_ID,
+        content = A_POLL_RESPONSE_CONTENT.toContent()
+)
+
+private val A_POLL_END_EVENT = Event(
+        type = EventType.POLL_END.first(),
+        eventId = AN_EVENT_ID,
+        originServerTs = 1652435922563,
+        senderId = A_USER_ID_1,
+        roomId = A_ROOM_ID,
+        content = A_POLL_END_CONTENT.toContent()
+)
+
+private val A_TIMELINE_EVENT = TimelineEvent(
+        root = A_POLL_START_EVENT,
+        localId = 1234,
+        eventId = AN_EVENT_ID,
+        displayIndex = 0,
+        senderInfo = SenderInfo(A_USER_ID_1, "A_USER_ID_1", true, null)
+)
+
+private val A_POLL_RESPONSE_EVENT_WITH_A_WRONG_REFERENCE = A_POLL_RESPONSE_EVENT.copy(
+        content = A_POLL_RESPONSE_CONTENT
+                .copy(
+                        relatesTo = RelationDefaultContent(
+                                type = RelationType.REPLACE,
+                                eventId = null
+                        )
+                )
+                .toContent()
+)
+
+private val A_POLL_REPLACE_EVENT = A_POLL_START_EVENT.copy(
+        content = A_POLL_CONTENT
+                .copy(
+                        relatesTo = RelationDefaultContent(
+                                type = RelationType.REPLACE,
+                                eventId = AN_EVENT_ID
+                        )
+                )
+                .toContent()
+)
+
+private val A_BROKEN_POLL_REPLACE_EVENT = A_POLL_START_EVENT.copy(
+        content = A_POLL_CONTENT
+                .copy(
+                        relatesTo = RelationDefaultContent(
+                                type = RelationType.REPLACE,
+                                eventId = null
+                        )
+                )
+                .toContent()
+)
+
+private val A_POLL_REFERENCE_EVENT = A_POLL_START_EVENT.copy(
+        content = A_POLL_CONTENT
+                .copy(
+                        relatesTo = RelationDefaultContent(
+                                type = RelationType.REFERENCE,
+                                eventId = AN_EVENT_ID
+                        )
+                )
+                .toContent()
+)
+
+private val AN_INVALID_POLL_RESPONSE_EVENT = A_POLL_RESPONSE_EVENT.copy(
+        content = AN_INVALID_POLL_RESPONSE_CONTENT.toContent()
+)
+
+class PollAggregationProcessorTest {
+
+    private val pollAggregationProcessor: PollAggregationProcessor = DefaultPollAggregationProcessor()
+    private val realm = FakeRealm()
+    private val session = mockk<Session>()
+
+    @Before
+    fun setup() {
+        mockEventAnnotationsSummaryEntity()
+        mockRoom(A_ROOM_ID, AN_EVENT_ID)
+        every { session.myUserId } returns A_USER_ID_1
+    }
+
+    @Test
+    fun `given a poll start event which is not a replace is not processed by poll aggregator`() {
```

Comment:
In the test names I think we should reflect the expected outcome, not simply the condition setup.

Following a given-when-then structure, I think the then clause is the most important as it shows us what is expected (personally I always use when-then with given as an optional if there are multiple paths/inputs to a function, but curious to know how other people go about this)

Reply:
Done. Let's discuss this on Friday with team.